mesonbuild/dependencies/base.py

```diff
@@ -327,6 +333,66 @@ def __init__(self, type_name, environment, kwargs, language: T.Optional[str] = N
         HasNativeKwarg.__init__(self, kwargs)
         self.clib_compiler = detect_compiler(self.name, environment, self.for_machine, self.language)
 
+    def convert_path_posix_to_native(self, p):
+        return convert_path_posix_to_native(p)
+
+    def get_include_paths_from_suffix(self, include_suffix):
+        global msystem
```

Comment:
I'll look over this more, but you don't need global here. You only need global when you want to *assign* to something from the global namespace, like:

```python
value = None

def set_value(value):
    global value
    value = foo
```

Reply:
OK, this is also the place I need considerating sys_root and CMAKE_PREFIX_PATH, please give me suggestion or solution